From: Miaohe Lin <linmiaohe@huawei.com>
Subject: mm/memory-failure.c: fix wrong user reference report

The dirty swapcache page is still residing in the swap cache after it's
hwpoisoned.  So there is always one extra refcount for swap cache.

Link: https://lkml.kernel.org/r/20220228140245.24552-3-linmiaohe@huawei.com
Signed-off-by: Miaohe Lin <linmiaohe@huawei.com>
Cc: Naoya Horiguchi <naoya.horiguchi@nec.com>
Signed-off-by: Andrew Morton <akpm@linux-foundation.org>
---

 mm/memory-failure.c |    6 +-----
 1 file changed, 1 insertion(+), 5 deletions(-)

--- a/mm/memory-failure.c~mm-memory-failurec-fix-wrong-user-reference-report
+++ a/mm/memory-failure.c
@@ -983,7 +983,6 @@ static int me_pagecache_dirty(struct pag
 static int me_swapcache_dirty(struct page_state *ps, struct page *p)
 {
 	int ret;
-	bool extra_pins = false;
 
 	ClearPageDirty(p);
 	/* Trigger EIO in shmem: */
@@ -992,10 +991,7 @@ static int me_swapcache_dirty(struct pag
 	ret = delete_from_lru_cache(p) ? MF_FAILED : MF_DELAYED;
 	unlock_page(p);
 
-	if (ret == MF_DELAYED)
-		extra_pins = true;
-
-	if (has_extra_refcount(ps, p, extra_pins))
+	if (has_extra_refcount(ps, p, true))
 		ret = MF_FAILED;
 
 	return ret;
_
